src/nv_ingest/api/v2/ingest.py

```diff
@@ -0,0 +1,507 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-25, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+# pylint: skip-file
+
+from io import BytesIO
+from typing import Any, Dict, List
+import base64
+import copy
+import json
+import logging
+import os
+import time
+import uuid
+
+from fastapi import APIRouter, Request, Response
+from fastapi import HTTPException
+from fastapi.responses import StreamingResponse
+from opentelemetry import trace
+from redis import RedisError
+
+from nv_ingest.framework.schemas.framework_message_wrapper_schema import MessageWrapper
+from nv_ingest_api.util.service_clients.client_base import FetchMode
+
+# For PDF splitting
+import pypdfium2 as pdfium
+
+# Reuse V1 state constants and helper functions
+from ..v1.ingest import (
+    trace_id_to_uuid,
+    INGEST_SERVICE_T,
+    STATE_RETRIEVED_DESTRUCTIVE,
+    STATE_RETRIEVED_NON_DESTRUCTIVE,
+    STATE_RETRIEVED_CACHED,
+    STATE_FAILED,
+    STATE_SUBMITTED,
+    INTERMEDIATE_STATES,
+)
+
+logger = logging.getLogger("uvicorn")
+tracer = trace.get_tracer(__name__)
+
+router = APIRouter()
+
+DEFAULT_PDF_SPLIT_PAGE_COUNT = 10
+
+
+def get_pdf_split_page_count() -> int:
+    """Resolve the configured page chunk size for PDF splitting."""
+
+    raw_value = os.environ.get("PDF_SPLIT_PAGE_COUNT")
+    if raw_value is None:
+        return DEFAULT_PDF_SPLIT_PAGE_COUNT
+
+    try:
+        parsed = int(raw_value)
+    except ValueError:
+        logger.warning(
+            "Invalid PDF_SPLIT_PAGE_COUNT '%s'; falling back to default %s", raw_value, DEFAULT_PDF_SPLIT_PAGE_COUNT
+        )
+        return DEFAULT_PDF_SPLIT_PAGE_COUNT
+
+    if parsed <= 0:
+        logger.warning("PDF_SPLIT_PAGE_COUNT must be >= 1; received %s. Using 1.", parsed)
+        return 1
+
+    return parsed
+
+
+def split_pdf_to_chunks(pdf_content: bytes, pages_per_chunk: int) -> List[Dict[str, Any]]:
+    """
+    Split a PDF into multi-page chunks using pypdfium2.
+
+    Returns a list of dictionaries containing the chunk bytes and page range metadata.
+    Note: this currently buffers each chunk in-memory; consider streaming in future upgrades.
+    """
+
+    chunks: List[Dict[str, Any]] = []
+
+    if pages_per_chunk <= 0:
+        pages_per_chunk = 1
+
+    pdf = pdfium.PdfDocument(pdf_content)
+    total_pages = len(pdf)
+
+    try:
+        for chunk_index, start_zero in enumerate(range(0, total_pages, pages_per_chunk)):
+            end_zero = min(start_zero + pages_per_chunk, total_pages)
+            page_indices = list(range(start_zero, end_zero))
+
+            new_pdf = pdfium.PdfDocument.new()
```

Comment:
I think when we do this, we're going to lose meta information on all the child PDF's from the parent. Not 100%, but we should probably check.

Reply:
They could probably be taken from the parent PDF and set on the child.

That said, if this is all we're losing, it may not be necessary.